pkg/util/network/network.go

```diff
@@ -0,0 +1,20 @@
+package network
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/feature"
+)
+
+func DetermineUserDefinedRouting(oc *api.OpenShiftCluster, sub *api.Subscription) error {
+	// Determine if this is a cluster with user defined routing
+	oc.Properties.NetworkProfile.OutboundType = api.OutboundTypeLoadbalancerRouting
+	if oc.Properties.APIServerProfile.Visibility == api.VisibilityPrivate &&
+		oc.Properties.IngressProfiles[0].Visibility == api.VisibilityPrivate &&
+		feature.IsRegisteredForFeature(sub.Properties, api.FeatureFlagUserDefinedRouting) {
+		oc.Properties.NetworkProfile.OutboundType = api.OutboundTypeUserDefinedRouting
+	}
+	return nil
+}
```

Comment:
We should move this to the installation package.  This isn't really a utility.  As well, we can define it on the manager struct and call it using `m.determineOutboundType`